src/librustc_mir/transform/inline.rs

```diff
@@ -692,12 +692,16 @@ impl<'a, 'tcx> MutVisitor<'tcx> for Integrator<'a, 'tcx> {
                 // Return pointer; update the place itself
                 *place = self.destination.clone();
             },
-            Place::Base(PlaceBase::Promoted(ref mut promoted)) => {
-                if let Some(p) = self.promoted_map.get(promoted.0).cloned() {
-                    promoted.0 = p;
+            Place::Base(PlaceBase::Static(ref mut static_)) => {
```

Comment:
Also directly destructure here instead of nesting a match

Reply:
I can destructure like this: `box mir::Static{promoted:Some(promoted), ..}`. But how do I change this? 
https://github.com/rust-lang/rust/blob/099177666c22bc1694d96804ad993f227ef3e579/src/librustc_mir/transform/inline.rs#L699